sssd_test_framework/utils/virtual_smartcard.py

```diff
@@ -0,0 +1,108 @@
+from pytest_mh import MultihostHost
+from pytest_mh.utils.fs import LinuxFileSystem
+
+__all__ = [
+    "SmartCardUtils",
+]
+class SmartCardUtils:
+    """
+    Utility class for managing smart card operations using SoftHSM and PKCS#11.
+    """
+    
+    def __init__(self, host: MultihostHost):
+        """
+        Initializes the SmartCardUtils with a given host.
+        
+        :param host: The remote host where smart card operations will be executed.
+        """
+        self.host = host
+        self.fs = LinuxFileSystem(host.conn)
+    
+    def init(self, label: str, so_pin: str, user_pin: str) -> None:
+        """
+        Initializes a SoftHSM token with the given label and PINs.
+        
+        :param label: The label for the new token.
+        :param so_pin: Security Officer PIN.
+        :param user_pin: User PIN.
+        """
+        self.host.conn.exec([
+            "softhsm2-util", "--init-token",
+            "--label", label,
+            "--free",
+            "--so-pin", so_pin,
+            "--pin", user_pin
+        ])
```

Comment:
In order for the virtual smart card setup to work, we need to make sure that softhsm2-util and libsofthsm2.so are using the correct config file by adding it like this:
```
        self.host.conn.exec([
            "softhsm2-util", "--init-token",
            "--label", "sc_test",
            "--free",
            "--so-pin", self.so_pin,
            "--pin", self.user_pin
        ],
        env={"SOFTHSM2_CONF": "/opt/test_ca/softhsm2.conf"})
```

So you should just need to add `env={"SOFTHSM2_CONF": "/opt/test_ca/softhsm2.conf"}` to the args for each `host.conn.exec()`